src/editors/data/redux/requests/reducer.test.js

```diff
@@ -0,0 +1,32 @@
+import { initialState, actions, reducer } from './reducer';
+import { RequestStates, RequestKeys } from '../../constants/requests';
+
+const testingState = {
+  ...initialState,
+  arbitraryField: 'arbitrary',
+};
+describe('requests reducer', () => {
+  test('intial state generated on create', () => {
+    expect(reducer(undefined, {})).toEqual(initialState);
+  });
+  describe('handling actions', () => {
+    const requestsList = ['fetchUnit', 'fetchBlock', 'saveBlock'];
+    const createTestParams = (requestKey) => [
+      [requestKey, 'startRequest', requestKey, { status: RequestStates.pending }],
+      [requestKey, 'completeRequest', { requestKey }, { status: RequestStates.completed, response: undefined }],
+      [requestKey, 'failRequest', { requestKey }, { status: RequestStates.failed, error: undefined }],
+      [requestKey, 'clearRequest', { requestKey }, {}],
+    ];
+    const lifecycletest = (target, action, payload, testValue) => {
```

Comment:
I would give this a slightly clearer name (testLifecycleActions), and have the foreach list included HERE rather than in the createTestParams.